restcomm/restcomm.mscontrol.jsr309/src/main/java/org/restcomm/connect/mscontrol/jsr309/Jsr309CallController.java

```diff
@@ -496,7 +496,7 @@ private void onCreateMediaSession(CreateMediaSession message, ActorRef self, Act
     }
 
     private void onCloseMediaSession(CloseMediaSession message, ActorRef self, ActorRef sender) throws Exception {
-        if (is(active) || is(initializing) || is(updatingMediaSession)) {
+        if (is(active) || is(initializing) || is(updatingMediaSession) || is(pending)) {
```

Comment:
is this "pending"  related to new flag "msResponsePending" ? i was expecting new flag to be here instead of this "pending"... can you clarify relation?

Reply:
i see this is different actor, is msResponsePending in voiceInterpreter actor, equivalent to "pending" in jsr309CallController?

Reply:
Not exactly. The boolean `msResponsePending` controls the sequence of communication between RC and a MS, while `is(pending)` verification allows the transition from `pending` to `inactive`, properly canceling a call branch. This case is particularly used when a dial fork occurs and a branch that was ringing (pending) answers back with BUSY, so we have to change to inactive. (You can see Henrique's suggestion on this here https://telestax.slack.com/archives/C0D8Y2AAG/p1481801494001255)